File: contains_duplicate.py
def containsduplicate(nums):
    result = set()
    for i in nums:
        if i in result:
            return True
        result.add(i)
    return False

print(containsduplicate( [1,2,3,1]))
print(containsduplicate( [1,2,3,4]))
print(containsduplicate( [1,1,1,3,3,4,3,2,4,2]))

File: containsDuplicate.cpp
#include<iostream>
#include<vector>
#include<map>
#include<unordered_set>
using namespace std;

bool ContainsDuplicate (vector<int> & nums) {
    unordered_set <int> Set;
    for (int i=0; i<nums.size(); i++) {
        if (Set.count(nums[i])) {
            return true;
        }
        Set.insert(nums[i]);
    }
    return false;
}

int main() {
    vector <int> nums;
    bool result;
    nums.push_back(1);
    nums.push_back(2);
    nums.push_back(3);
    nums.push_back(4);
 
    result = ContainsDuplicate (nums);
    cout << "results "<< result << endl;
    //clear the vector 
    nums.clear();
    nums.push_back(1);
    nums.push_back(2);
    nums.push_back(3);
    nums.push_back(1);
    

    result = ContainsDuplicate (nums);
    cout << "results "<< result << endl;
    
    return 0;
}
